### PR TITLE
[Proposal] Update to package version 2.3.0

### PR DESCRIPTION
**Pull request by B1TC0R3**
Hello Danny,

cool template overall! Made some changes, hope you like them.

## What changed?

- I moved the language handling to it's own file so it becomes easier to manage and add new ones when necessary. Also makes the code a lot more readable in my opinion.
- I also refactored some of the source files, I hope that is fine.

In total, the look of the template should **not** change much, only the source code.
Please let me know if I screwed something up so I can fix it. :smiley: 

## Important

I already updated the version number in the `.toml` file, but you will need to actually compress the changes into a new package version and sync it with Typst yourself, if you want to accept my changes.

Have a nice evening, :wave:
Thomas.

### CODE REVIEW

**Comment by DannySeidel** [comment]
Hi Thomas,

Thanks for the ideas. I really like some of the changes you made, especially the idea of putting the language handling in a separate file, which makes a lot of sense. Since there are no new features, I don't see any need to quickly merge the changes. I will incorporate them into my next release.

Thanks again 🙏🏻

File: locale.typ [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+#let DECLARATION_OF_OWNERSHIP_SECTION_B_SINGLE = (
+  "de": "selbstständig verfasst und keine anderen als die angegebenen Quellen und Hilfsmittel benutzt habe. Ich versichere zudem, dass die eingereichte elektronische Fassung mit der gedruckten Fassung übereinstimmt.",
+  "en": "independently and have not used any sources or aids other than those specified. I also confirm that the electronic version submitted is identical to the printed version."
+)
+
+#let DECLARATION_OF_OWNERSHIP_SECTION_B_PLURAL = (
+  "de": "selbstständig verfasst und keine anderen als die angegebenen Quellen und Hilfsmittel benutzt haben. Wir versichern zudem, dass die eingereichte elektronische Fassung mit der gedruckten Fassung übereinstimmt.
+",
+  "en": "independently and have not used any sources or aids other than those specified. We also confirm that the electronic version submitted is identical to the printed version.",
+)
```
Comment (DannySeidel) [inline locale.typ:60]: The declaration of authorship is required to be in German according to the DHBW guidelines, so the English version is not needed here.

File: titlepage.typ [hunk 2]
```diff
@@ -20,7 +21,9 @@
   confidentiality-marker,
   university-short,
 ) = {
-  if (many-authors) {
+  let many_authors = authors.len() > 3
```
Comment (DannySeidel) [inline titlepage.typ:24]: kebab-case again 😉

**Comment by DannySeidel** [comment]
@B1TC0R3 you don't have to change any of this. I just wanted to let you know what I think about different things.

File: confidentiality-statement.typ [hunk 3]
```diff
@@ -64,14 +61,14 @@
...
-  }])
+  text(
+    authors_by_city.dedup().join(", ", last: AND.at(lang)) +
+    [ ] +
+    end-date.display(date-format)
+  )
 
   v(0.5em)
-  if (many-authors) {
+  if (authors.len() > 3) {
```
[the remark this answers is not in the document]
Comment (B1TC0R3) [reply confidentiality-statement.typ:71]: It was only used multiple times in a single file, so I replaced the extra "multiple-authors" parameter for better readability.
On the titlepage, where it was used multiple times, I added a variable for it at the very top.

File: locale.typ (hunk 1 above)
Comment (B1TC0R3) [reply locale.typ:60]: In this case, should I hard-code it to use the German version?
Currently, it uses whatever language the user sets in the configuration.

**Comment by B1TC0R3** [comment]
I will see that I fix what you pointed out later today.
We do adhere to coding standards here, after all. ;)

File: confidentiality-statement.typ (hunk 3 above)
[the remark this answers is not in the document]
Comment (DannySeidel) [reply confidentiality-statement.typ:71]: I get what you are trying to do, but if I want to change the value at some point, I only want to change it in one place, not three. Therefore, I had the parameter defined in the `lib` file to set the central value there.

**Comment by DannySeidel** [comment]
I already have a PR containing your changes here https://github.com/typst/packages/pull/865. So no need to change anything.

File: locale.typ (hunk 1 above)
Comment (DannySeidel) [reply locale.typ:60]: See how I changed it here https://github.com/typst/packages/pull/865.

But I think it makes sense to have a way of overwriting the default as you say.

Comment (DannySeidel) [reply locale.typ:60]: I already have a parameter in the template to provide a custom confidentiality statement. I think it makes sense to add a parameter to overwrite the whole declaration of authorship to provide your own custom one in the same way.

**Comment by B1TC0R3** [comment]
I see.

In this case you can probably close this pull request.

**Comment by DannySeidel** [comment]
I will close it once the changes are merged 👍🏻

**Pull request closed without merging**